ImageGallery/src/org/sleuthkit/autopsy/imagegallery/datamodel/grouping/GroupManager.java

```diff
@@ -618,14 +624,65 @@ synchronized public void handleFileUpdate(Collection<Long> updatedFileIDs) {
             for (GroupKey<?> gk : groupsForFile) {
                 // see if a group has been created yet for the key
                 DrawableGroup g = getGroupForKey(gk);
+               
+                checkForPathGroupChange(gk);
                 addFileToGroup(g, gk, fileId);
             }
         }
 
         //we fire this event for all files so that the category counts get updated during initial db population
         controller.getCategoryManager().fireChange(updatedFileIDs, null);
     }
+    
+    /**
+     * Checks if the given path is different from the last updated path group.
+     * 
+     * The idea is that when the path of the files being processed changes, 
+     * we have moved from one folder to the next, and the group for the 
+     * previous PATH can be considered as analyzed and can be displayed.
+     * 
+     * NOTE: this a close approximation for when all files in a folder have been processed, 
+     * but there's some room for error - files may go down the ingest pipleline  
+     * out of order or the events may not always arrive in the same order
+     * 
+     * @param groupKey 
+     */
+    private void checkForPathGroupChange(GroupKey<?> groupKey) {
```

Comment:
In my opinion, the way this method is documented and named, the changing of the filed tracking the last updated path group is a side effect - everything is about checking but not abut changing, e.g. "Checks if the given path is different from the last updated path group."  I suggest that we rename this method to something like updateLastUpdatedPathGroup. Or perhaps the field lastUpdatedPathGroup is actually something more like currentPathGroup and the method could be updateCurrentPathGroup? Does that express the concept better?